caikit/runtime/servicers/runtime_info_servicer.py

```diff
@@ -0,0 +1,68 @@
+# Copyright The Caikit Authors
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+# Have pylint ignore Class XXXX has no YYYY member so that we can use gRPC enums.
+# pylint: disable=E1101
+
+# Standard
+from typing import Dict
+from importlib.metadata import version as pip_version
+
+# Third Party
+import grpc
+
+# First Party
+import alog
+
+# Local
+from caikit.core import MODEL_MANAGER
+from caikit.interfaces.runtime.data_model import (
+    RuntimeInfoStatusResponse,
+    RuntimeInfoStatusResponseDict,
+)
+from caikit.config import get_config
+
+log = alog.use_channel("RI-SERVICR-I")
+
+
+class RuntimeInfoServicerImpl:
+    """This class contains the implementation of all of the RPCs that are required to run a
+    service in Model Mesh as a Model-Runtime."""
+
+    def GetRuntimeInfo(self, request, context):
```

Comment:
What is the `request` and `context` here? It is needed in order for the GRPC tests to run otherwise I get error
```
ERROR    grpc._server:_server.py:580 Exception calling application: RuntimeInfoServicerImpl.GetRuntimeInfo() takes 1 positional argument but 3 were given
Traceback (most recent call last):
  File "/Users/anhuong/github.com/anhuong/caikit/.tox/py/lib/python3.11/site-packages/grpc/_server.py", line 552, in _call_behavior
    response_or_iterator = behavior(argument, context)
                           ^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/anhuong/github.com/anhuong/caikit/.tox/py/lib/python3.11/site-packages/py_grpc_prometheus/prometheus_server_interceptor.py", line 118, in new_behavior
    raise e
  File "/Users/anhuong/github.com/anhuong/caikit/.tox/py/lib/python3.11/site-packages/py_grpc_prometheus/prometheus_server_interceptor.py", line 67, in new_behavior
    response_or_iterator = behavior(request_or_iterator, servicer_context)
                           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/anhuong/github.com/anhuong/caikit/caikit/runtime/interceptors/caikit_runtime_server_wrapper.py", line 149, in safe_rpc_call
    return rpc(request, context)
           ^^^^^^^^^^^^^^^^^^^^^
TypeError: RuntimeInfoServicerImpl.GetRuntimeInfo() takes 1 positional argument but 3 were given
```

but when I run with request and context, HTTP server fails with error: `TypeError: RuntimeInfoServicerImpl.GetRuntimeInfo() missing 2 required positional arguments: 'request' and 'context'`

Reply:
`request`(protobuf message defined for the endpoint)  and `context` (grpc context) are both necessary for GRPC endpoint handling

Reply:
@anhuong Since this is a grpc endpoint handler, I probably wouldn't try to call it directly from an http endpoint handler. It should be simple to refactor the main logic out of this and into a separate public method that both the grpc and http endpoint handlers can call.

As for the `context`, it's suuuuuper hard to build a real one of those objects yourself, they're an implementation detail of the grpc server itself. But we do have a test fixture that can give you a fake one: https://github.com/caikit/caikit/blob/main/tests/fixtures/fixtures.py#L42